FirebaseMessaging/Tests/IntegrationTests/FIRMessagingTokenRefreshTests.swift

```diff
@@ -97,6 +96,30 @@
       wait(for: [expectation, notificationExpectation], timeout: 5)
     }
 
+    func testDeleteDataWithTokenRefreshDelegatesAndNotifications() {
+      let expectation = self.expectation(description: "delegate method and notification are called")
+      assertDefaultToken()
+
+      let notificationExpectation = self.expectation(forNotification: NSNotification.Name
+        .MessagingRegistrationTokenRefreshed,
+        object: nil,
+        handler: nil)
+
+      let testDelegate = fakeAppDelegate()
+      messaging?.delegate = testDelegate
+      testDelegate.delegateIsCalled = false
+
+      guard let messaging = self.messaging else {
+        return
+      }
```

Comment:
I think it'd be helpful to ensure that the `messaging` property is non-nil by the time each test case runs. This prevents need for code in each test case that works around the fact that `messaging` is an optional property.

- Declare the `messaging` as explicitly unwrapped
- Use [setUpWithError](https://developer.apple.com/documentation/xctest/xctest/3521150-setupwitherror?language=objc) API to assign it

Here's what the refactor would look like:
```diff
 class FIRMessagingTokenRefreshTests: XCTestCase {
    var app: FirebaseApp!
-    var messaging: Messaging?
+    var messaging: Messaging!
    override class func setUp() {
      FirebaseApp.configure()
    }
-    override func setUp() {
+   override func setUpWithError() throws {
      messaging = Messaging.messaging()
+    messaging = try XCTUnwrap(Messaging.messaging())
    }
```

With this refactor, the optional related code in the tests can be removed. For this case:

```suggestion
      let testDelegate = fakeAppDelegate()
      messaging.delegate = testDelegate
      testDelegate.delegateIsCalled = false
```

The same refactor can be applied to FirebaseMessaging/Tests/IntegrationTests/FIRMessagingPubSubTest.swift to simplify things as well.

Reply:
Done. Thanks!